src/completion.ts

```diff
@@ -104,18 +105,24 @@ export function asCompletionItem(entry: tsp.CompletionEntry, file: string, posit
             }
         }
     }
+    if (!insertText) {
+        insertText = item.label;
+    }
     if (replacementRange) {
-        if (!insertText) {
-            insertText = item.label;
-        }
-
-        item.textEdit = lsp.TextEdit.replace(replacementRange, insertText);
-    } else {
-        item.insertText = insertText;
+        item.textEdit = createTextEdit(replacementRange, position, insertText, features);
     }
+    item.insertText = insertText;
```

Comment:
Should I put `item.insertText`  back in an else branch?

Reply:
We should keep the logic as close to how it was before as possible.
Maybe this change is the reasons the tests are failing even?